src/main/java/org/spongepowered/api/world/Location.java

```diff
@@ -666,24 +666,24 @@ public Entity createEntity(EntityType type) {
      * should be taken note that there can be many reasons for a failure.</p>
      *
      * @param entity The entity to spawn
-     * @return True if successful, false if not
+     * @return The entity, if it was spawned successfully
      * @see EntityUniverse#spawnEntity(Entity)
      */
-    public boolean spawnEntity(Entity entity) {
+    public Optional<Entity> spawnEntity(Entity entity) {
         return this.getExtent().spawnEntity(entity);
     }
 
     /**
-     * Similar to {@link #spawnEntity(Entity, Cause)} except where multiple
+     * Similar to {@link #spawnEntity(Entity)} except where multiple
      * entities can be attempted to be spawned with a customary {@link Cause}.
      * The recommended use is to easily process the entity spawns without
      * interference with the cause tracking system.
      *
      * @param entities The entities to be spawned
-     * @return True if any of the entities were successfully spawned
-     * @see EntityUniverse#spawnEntities(Iterable, Cause)
+     * @return The entities which spawned correctly, or empty if none
+     * @see EntityUniverse#spawnEntities(Iterable)
      */
-    public boolean spawnEntities(Iterable<? extends Entity> entities) {
+    public Collection<Entity> spawnEntities(Iterable<? extends Entity> entities) {
```

Comment:
Now this one, I can see the reasoning behind. Else you can't know who spawned without a way to check each Entity (which makes things somewhat annoying).

Reply:
This one is most likely less commonly used as well, so it hurts less to break.